client/signup/jetpack-connect/index.jsx

```diff
@@ -199,6 +199,10 @@ class JetpackConnectMain extends Component {
 			return false;
 		}
 
+		if ( this.checkProperty( 'isJetpackConnected' ) ) {
```

Comment:
> Thinking out loud and this is outside the scope of this PR, but it would be nice to rethink this whole section.

:+1: agree, but let's keep it for another PR 😉 

> Is the reordering necessary? When working on #13500 I noticed that these strings which are passed to JetpackConnectNotices do not all match. See bb9e1e5

Yes, because `alreadyOwned` will redirect the user to plans. Although I don't 100% remember, there was some rationale to keep that separate. Maybe @johnHackworth can shed some light on that.